src/Dumplie/Domain/SharedKernel/Money/Price.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Domain\SharedKernel\Money;
+
+use Dumplie\Domain\SharedKernel\Exception\DifferentPricePrecisionException;
+use Dumplie\Domain\SharedKernel\Exception\InvalidArgumentException;
+use Dumplie\Domain\SharedKernel\Exception\InvalidCurrencyException;
+
+final class Price
+{
+    const DEFAULT_PRECISION = 100;
+
+    /**
+     * @var int
+     */
+    private $amount;
+
+    /**
+     * @var string
+     */
+    private $currency;
+
+    /**
+     * @var int
+     */
+    private $precision;
+
+    /**
+     * @param int    $amount    Amount, expressed in the smallest units of $currency (eg cents)
+     * @param string $currency
+     * @param int    $precision used to calculate float value, 10000 cents / 100 = 100.00
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct(int $amount, string $currency, int $precision = self::DEFAULT_PRECISION)
+    {
+        if ($amount < 0 || $precision < 0) {
+            throw new InvalidArgumentException();
+        }
```

Comment:
I would do something like:

```php
if ($amount < 0) {
    throw new InvalidArgumentException('Amount cannot be less than 0');
}
if ($amount < 0) {
    throw new InvalidArgumentException('Amount cannot be less than 0');
}
if ($precision < 0) {
    throw new InvalidArgumentException('Precision cannot be less than 0');
}
```

so during debuging or some issues we would now what happened